components/shared/ProgressBar.tsx

```diff
@@ -0,0 +1,104 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+// Modified code from https://github.com/MatinAniss/traction.js
+
+interface ProgressBarProps {
+  color: string;
+  incrementInterval: number;
+  incrementAmount: number;
+  transitionDuration: number;
+  transitionTimingFunction:
+    | "ease"
+    | "linear"
+    | "ease-in"
+    | "ease-out"
+    | "ease-in-out";
+}
+
+export const ProgressBar: React.FC<ProgressBarProps> = (props) => {
+  // Declare states
+  const router = useRouter();
+  const [progress, setProgress] = useState(0);
+  const [isVisible, setIsVisible] = useState(false);
+
+  // Progress bar inline styling
+  const styling = {
+    position: "fixed",
+    top: 0,
+    left: 0,
+    width: `${progress}%`,
+    height: "3px",
+    backgroundColor: props.color,
+    transition: `width ${props.transitionDuration}ms ${props.transitionTimingFunction}`,
+    opacity: isVisible ? 1 : 0,
+    zIndex: 9999999999,
+  } as React.CSSProperties;
+  useEffect(() => {
+    // Declare timeout
+    let status: "in-progress" | "idle" = "idle";
+    let intervalId: NodeJS.Timeout;
+
+    // Route change start function
+    const onRouteChangeStart = () => {
+      status = "in-progress";
+      setIsVisible(true);
```

Comment:
can we delay this by let's say 100ms or 200ms so that if the route change completes before we don't show it?